models/resnet/resnet.hpp

```diff
@@ -0,0 +1,554 @@
+    inputWidth = ConvOutSize(inputWidth, kernelWidth, strideWidth, padW);
+    inputHeight = ConvOutSize(inputHeight, kernelHeight, strideHeight, padH);
+
+    mlpack::Log::Info << outSize << ", " << inputWidth << ", " << inputHeight
+        << ")" << std::endl;
+  }
+
+  /**
+   * Adds a 1x1 Convolution Block.
+   *
+   * @tparam SequentialType Layer type in which 1x1 convolution block will
+   *     be added.
+   *
+   * @param inSize Number of input maps.
+   * @param outSize Number of output maps.
+   * @param downSampleInputWidth Input widht for downSample block.
+   * @param downSampleInputHeight Input height for downSample block.
+   * @param strideWidth Stride of filter application in the x direction.
+   * @param strideHeight Stride of filter application in the y direction.
+   * @param kernelWidth Width of the filter/kernel.
+   * @param kernelHeight Height of the filter/kernel.
+   * @param padW Padding width of the input.
+   * @param padH Padding height of the input.
+   * @param downSample Bool if it's a downsample block or not. Default is false.
+   */
+  template<typename SequentialType = ann::Sequential<>>
+  void ConvolutionBlock1x1(SequentialType* baseLayer,
+                           const size_t inSize,
+                           const size_t outSize,
+                           const size_t downSampleInputWidth = 0,
+                           const size_t downSampleInputHeight = 0,
+                           const size_t strideWidth = 1,
+                           const size_t strideHeight = 1,
+                           const size_t kernelWidth = 1,
+                           const size_t kernelHeight = 1,
+                           const size_t padW = 0,
+                           const size_t padH = 0,
+                           const bool downSample = false)
+  {
+    if (downSample)
+    {
+      baseLayer->Add(new ann::Convolution<>(inSize, outSize, kernelWidth,
+          kernelHeight, strideWidth, strideHeight, padW, padH,
+          downSampleInputWidth, downSampleInputHeight));
+
+      mlpack::Log::Info << "  Convolution: " << "(" << inSize << ", " <<
+          downSampleInputWidth << ", " << downSampleInputHeight << ")" <<
+          " ---> (" << outSize << ", " << downSampleInputWidth << ", " <<
+          downSampleInputHeight << ")" << std::endl;
+    }
+    else
+    {
+      baseLayer->Add(new ann::Convolution<>(inSize, outSize, kernelWidth,
+          kernelHeight, strideWidth, strideHeight, padW, padH,
+          inputWidth, inputHeight));
+
+      mlpack::Log::Info << "Convolution: " << "(" << inSize << ", " <<
+          inputWidth << ", " << inputHeight << ")" << " ---> (";
+
+      // Updating input dimensions.
+      inputWidth = ConvOutSize(inputWidth, kernelWidth, strideWidth, padW);
+      inputHeight = ConvOutSize(inputHeight, kernelHeight, strideHeight, padH);
+
+      mlpack::Log::Info << outSize << ", " << inputWidth << ", " << inputHeight
+          << ")" << std::endl;
+    }
```

Comment:
Lets also avoid else statements.

Reply:
Not sure how can i go by it then, because there isn't a condition that always executes.

Reply:
Umm, return statement inside first `if` condition?

Reply:
btw for this part can you take a look at the new change? i think would need to add batch norm in the if statment too if i remove else, right ?